readme.md

```diff
@@ -1,161 +1,181 @@
+    to remotely manage the city lights.
+  license:
+    name: Apache 2.0
+    url: 'https://www.apache.org/licenses/LICENSE-2.0'
+
+defaultContentType: application/json
+
+servers:
+  local:
+    url: localhost:5672
+    protocol: amqp
+    security:
+      - user-password: []
+
+channels:
+  light/measured:
+    bindings:
+      amqp:
+        is: routingKey
+        queue:
+          name: light/measured
+          durable: true
+          exclusive: true
+          autoDelete: false
+          vhost: /
+        bindingVersion: 0.2.0
+    publish:
+      summary: Inform about environmental lighting conditions for a particular streetlight.
+      operationId: onLightMeasured
+      message:
+        name: LightMeasured
+        payload:
+          $id: LightMeasured
+          additionalProperties: false
+          type: object
+          properties:
+            id:
+              type: integer
+              minimum: 0
+              description: Id of the streetlight.
+            lumens:
+              type: integer
+              minimum: 0
+              description: Light intensity measured in lumens.
+            sentAt:
+              type: string
+              format: date-time
+              description: Date and time when the message was sent.
+
+components:
+  securitySchemes:
+    user-password:
+      type: userPassword
+```
 
-## What is it?
-AsyncAPI To-Go is a code generation template for the [AsyncAPI generator](https://github.com/asyncapi/generator) that can be used to generate boilerplate Go code for interacting with event-driven architectures. 
+#### Run the following command to generate a Go module
 
-The generator consumes an [**AsyncAPI 2.0**](https://www.asyncapi.com/docs/specifications/2.0.0/) YAML or JSON specification file and uses this template to produce matching channel constants, message payload structs, and (W.I.P.) ~~optionally, logic to wrap it into a lightweight library.~~ The underlying transport layer interface enables pluggable support for most messaging system (Kafka, MQTT, AMPQ, etc.).
+```bash
+npm install -g @asyncapi/generator
+# clone this repository and navigate to this repository
+ag /path/to/asyncapi.yaml ./ -o /path/to/generated-code -p moduleName=your-go-module-name -p goVersion=1.17
+```
 
-The template provides switches to allow for different levels of generation. More information can be found at [generation levels](#generation-levels)
-* Channel parsing / generation with parameter structs
-* Schema and Message definitions
-* ~~Content Type parsing with the ability to extend supported content types~~ (Soon)
-* A transport interface to support various underlying event messaging protocols
-* ~~A simple wrapping framework to provide AsyncAPI *operations* as high-level functions~~ (Very Soon)
+There are 2 options that can be passed to the generator
 
-## What Am I Getting Myself Into? 
-Code Samples:
+1. moduleName: name of the go module to be generated
```

Comment:
It should be in section like https://github.com/asyncapi/nodejs-template#template-configuration that you already have mentioned in the TOC